src/py42/services/_connection.py

```diff
@@ -76,6 +77,24 @@ def get_host_address(self):
         return self._kv_service.get_stored_value(self._key).text
 
 
+class ConnectedServerHostResolver(HostResolver):
+    """A connection used in Push Restores to verify the accepting device is connected
+    to the Authority server."""
+
+    def __init__(self, connection, device_guid):
+        self._connection = connection
+        self._device_guid = device_guid
+        super(ConnectedServerHostResolver, self).__init__()
+
+    def get_host_address(self):
+        response = self._connection.get(
+            u"api/connectedServerUrl", params={u"guid": self._device_guid}
+        )
+        if response[u"serverUrl"] is None:
```

Comment:
Response when offine is like `{"serverUrl": null}` which causes the logic for checking if the host address begins with the `http` or `https` to fail with `None has no method startsWith()` which is confusing, so this custom exception is quite helpful